microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -302,38 +301,89 @@ def _kdump_test(self, node: Node, log_path: Path, log: Logger) -> None:
         except Exception as identifier:
             log.debug(f"ignorable ssh exception: {identifier}")
 
-        # Wait for the VM dump file and boot up
-        # When the crash kernel boots up, port 22 can be connected even if dump file
-        # is not completed. So we can't use wait_tcp_port_ready function to judge if
-        # the dump process is completed and the VM already boots up. We can use
-        # try_connect function
+        # Check if the vmcore file is generated after triggering a crash
+        self._check_kdump_result(node, log_path, log, kdump)
+
+    def _check_kdump_result(
+        self, node: Node, log_path: Path, log: Logger, kdump: KdumpBase
+    ) -> None:
         timer = create_timer()
         remote_node = cast(RemoteNode, node)
-        stdout = None
-        while not stdout and timer.elapsed(False) < self.timeout_of_dump_crash:
+        system_disconnected = True
+        serial_console = node.features[SerialConsole]
+        while system_disconnected and timer.elapsed(False) < self.timeout_of_dump_crash:
             try:
-                stdout = try_connect(remote_node._connection_info)
+                try_connect(remote_node._connection_info)
             except Exception as identifier:
                 log.debug(
-                    "failed to connect SSH "
+                    "Fail to connect SSH "
                     f"{remote_node._connection_info.address}:"
                     f"{remote_node._connection_info.port}. "
                     f"{identifier.__class__.__name__}: {identifier}. Retry..."
                 )
-                time.sleep(10)
-            if not stdout:
-                serial_console = node.features[SerialConsole]
                 serial_console.check_initramfs(
                     saved_path=log_path, stage="after_trigger_crash", force_run=True
                 )
-        if not stdout:
-            raise LisaException("Timeout to connect the VM after triggering kdump.")
+                system_disconnected = True
+                continue
 
-        # After trigger kdump, the VM will reboot. We need to close and re-initialize
-        node.close()
-        node.initialize()
-        # Verify if the kernel kdump process creates a vmcore file of size 10M+
-        kdump.check_vmcore_exist()
+            # If there is no exception, then the system is connected
+            system_disconnected = False
+
+            # After trigger kdump, the VM will reboot. We need to close and initialize
+            node.close()
+            node.initialize()
```

Comment:
no need run initialize. after close, initialize will be invoked when run any commands against the node

Reply:
Have deleted it.